pkg/build/controller/common/util.go

```diff
@@ -42,6 +48,85 @@ func HandleBuildCompletion(build *buildapi.Build, runPolicies []policy.RunPolicy
 	if err := runPolicy.OnComplete(build); err != nil {
 		utilruntime.HandleError(fmt.Errorf("failed to run policy on completed build %s/%s: %v", build.Namespace, build.Name, err))
 	}
+	if err := HandleBuildPruning(build, buildLister, buildConfigGetter, buildDeleter); err != nil {
+		utilruntime.HandleError(fmt.Errorf("failed to prune old builds %s/%s: %v", build.Namespace, build.Name, err))
+	}
+}
+
+func isCompletedBuild(build buildapi.Build) bool {
+	return build.Status.Phase == buildapi.BuildPhaseComplete
+}
+
+func isFailedBuild(build buildapi.Build) bool {
+	return build.Status.Phase == buildapi.BuildPhaseFailed
+}
```

Comment:
i don't think it's worth introducing these helpers unless you're going to check multiple phases, but if you are set on keeping them, please move them into pkg/build/util.go where we already have some helpers for checking build phases.